File: app/src/main/java/com/billiamram/prioritizer/Module.java
package com.billiamram.prioritizer;

import com.billiamram.prioritizer.domain.ItemSetRepository;

public class Module {
    private static ItemSetListPresenter itemSetListPresenter;
    private static ItemSetDetailsPresenter itemSetDetailsPresenter;
    private static ItemSetRepository itemSetRepository = new DemoItemSetRepository();

    public static ItemSetListPresenter provideItemSetListPresenter() {
        return itemSetListPresenter != null ? itemSetListPresenter : new ItemSetListPresenter(itemSetRepository);
    }

    public static ItemSetDetailsPresenter provideItemSetDetailsPresenter() {
        return itemSetDetailsPresenter != null ? itemSetDetailsPresenter : new ItemSetDetailsPresenter(itemSetRepository);
    }
}

File: app/src/main/java/com/billiamram/prioritizer/Module.kt
package com.billiamram.prioritizer

object Module {
    private var itemSetListPresenter: ItemSetListPresenter? = null
    private var itemSetDetailsPresenter: ItemSetDetailsPresenter? = null
    private val itemSetRepository = DemoItemSetRepository()

    fun provideItemSetListPresenter(): ItemSetListPresenter {
        itemSetListPresenter = itemSetListPresenter ?: ItemSetListPresenter(itemSetRepository)
        return itemSetListPresenter!!
    }

    fun provideItemSetDetailsPresenter(): ItemSetDetailsPresenter {
        itemSetDetailsPresenter = itemSetDetailsPresenter ?: ItemSetDetailsPresenter(itemSetRepository)
        return itemSetDetailsPresenter!!
    }
}
